File: eBarberShop/eBarberShop.Model/IzvjestajRezervacije.cs
namespace eBarberShop.Model
{
    public class IzvjestajRezervacije
    {
        public int UkupnoRezervacija { get; set; }
        public int UkupnoAktivnihRezervacija { get; set; }
        public int UkupnoNeaktivnihRezervacija { get; set; }
        public decimal ZaradaPrethodnogDana { get; set; }
        public decimal ZaradaPrethodneSemice { get; set; }
        public decimal ZaradaPrethodnogMjeseca { get; set; }
        public int BrojRezervacijaPrethodnogDana { get; set; }
        public int BrojRezervacijaPrethodneSedmice { get; set; }
        public int BrojRezervacijaPrethodnogMjeseca { get; set; }
        public int BrojUsluga { get; set; }
        public string UposlenikSaNajviseRezervacija { get; set; }
        public string UslugaSaNajviseRezervacija { get; set; }
        public decimal UkupnaZarada { get; set; }
    }
}

File: eBarberShop/UI/ebarbershop_desktop/lib/models/rezervacije/izvjestaj/izvjestaj_rezervacije.dart
import 'package:json_annotation/json_annotation.dart';

part 'izvjestaj_rezervacije.g.dart';

@JsonSerializable()
class IzvjestajRezervacije {
  int ukupnoRezervacija;
  int ukupnoAktivnihRezervacija;
  int ukupnoNeaktivnihRezervacija;
  double zaradaPrethodnogDana;
  double zaradaPrethodneSemice;
  double zaradaPrethodnogMjeseca;
  int brojRezervacijaPrethodnogDana;
  int brojRezervacijaPrethodneSedmice;
  int brojRezervacijaPrethodnogMjeseca;
  int brojUsluga;
  String uposlenikSaNajviseRezervacija;
  String uslugaSaNajviseRezervacija;
  double ukupnaZarada;

  IzvjestajRezervacije(
      this.ukupnoRezervacija,
      this.ukupnoAktivnihRezervacija,
      this.ukupnoNeaktivnihRezervacija,
      this.zaradaPrethodnogDana,
      this.zaradaPrethodneSemice,
      this.zaradaPrethodnogMjeseca,
      this.brojRezervacijaPrethodnogDana,
      this.brojRezervacijaPrethodneSedmice,
      this.brojRezervacijaPrethodnogMjeseca,
      this.brojUsluga,
      this.uposlenikSaNajviseRezervacija,
      this.uslugaSaNajviseRezervacija,
      this.ukupnaZarada);

  factory IzvjestajRezervacije.fromJson(Map<String, dynamic> json) =>
      _$IzvjestajRezervacijeFromJson(json);

  Map<String, dynamic> toJson() => _$IzvjestajRezervacijeToJson(this);
}
